pkg/types/powervs/validation/machinepool.go

```diff
@@ -0,0 +1,67 @@
+package validation
+
+import (
+	"math"
+	"regexp"
+	"strconv"
+
+	"github.com/google/uuid"
+	"github.com/openshift/installer/pkg/types/powervs"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// ValidateMachinePool checks that the specified machine pool is valid.
+func ValidateMachinePool(p *powervs.MachinePool, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Validate VolumeIDs
+	for i, volumeID := range p.VolumeIDs {
+		_, err := uuid.Parse(volumeID)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("volumeIDs").Index(i), volumeID, "volume ID must be a valid UUID"))
+		}
+	}
+
+	// Validate Memory
+	if p.Memory != "" {
+		memory, err := strconv.ParseInt(p.Memory, 10, 64)
+		if err == nil {
+			if memory < 2 || memory > 64 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("memory"), p.Memory, "memory must be from 2 to 64 GB"))
+			}
+		} else {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("memory"), p.Memory, "memory must be an integer number of GB that is at least 2 and no more than 64"))
+		}
+	}
+
+	// Validqte Processors
+	if p.Processors != "" {
+		processors, err := strconv.ParseFloat(p.Processors, 64)
+		if err == nil {
+			if processors < 0.25 || processors > 32 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("processors"), p.Processors, "number of processors must be from .25 to 32 cores"))
+			}
+			if math.Mod(processors*1000, 2) == 0 && math.Mod(processors*100, 25) == 0 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("processors"), p.Processors, "processors must be in increments of .25"))
+			}
+		} else {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("processors"), p.Processors, "processors must be a valid floating point number"))
+		}
+	}
+	// Validate ProcType
+	if p.ProcType != "" {
+		if p.ProcType != "shared" && p.ProcType != "dedicated" && p.ProcType != "capped" {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("procType"), p.ProcType, "procType must be one of: 'capped', 'shared' or 'dedicated'"))
+		}
+	}
+
+	// Validate SysType
+	if p.SysType != "" {
+		const sysTypeRegex = `^(?:e980|s922(-.*|))$`
```

Comment:
are there concrete values here we could list rather than a regex? i mean i would assume there would be invalid values even in this case ?

Reply:
the dev team creates us special system pools sometimes in staging that will follow the pattern s922-* (so like, s922-ocp). if we don't account for that here we'd have to recompile with a specific system pool for testing in staging. (and it's never with the e980s since those are so large, expensive, and therefore few)

Reply:
oops. also, system type is really just another word for system pool in practicality but we shouldn't really use the two terms interchangeably so apologies if that was confusing

Reply:
ah i see..so these have to be pre-created before we start the install.